internal/feed/reddit.go

```diff
@@ -30,8 +30,13 @@ type subredditResponseJson struct {
 	} `json:"data"`
 }
 
-func FetchSubredditPosts(subreddit string, commentsUrlTemplate string) (ForumPosts, error) {
-	requestUrl := fmt.Sprintf("https://www.reddit.com/r/%s/hot.json", url.QueryEscape(subreddit))
+func FetchSubredditPosts(subreddit string, commentsUrlTemplate string, requestUrlTemplate string) (ForumPosts, error) {
+	subreddit = url.QueryEscape(subreddit)
+	requestUrl := fmt.Sprintf("https://www.reddit.com/r/%s/hot.json", subreddit)
+	if commentsUrlTemplate != "" {
+		strings.ReplaceAll(commentsUrlTemplate, "{REQUEST-URL}", subreddit)
```

Comment:
`strings.ReplaceAll` returns a new copy of the string with the replacements made so you'll have to assign its result to `requestUrl`.

```suggestion
		requestUrl = strings.ReplaceAll(requestUrlTemplate, "{REQUEST-URL}", subreddit)
```